delta/plugins/storage/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/FileFixtures.scala

```diff
@@ -11,46 +13,60 @@ import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.nxv
-  val orgDeprecated                 = Label.unsafe("org-deprecated")
-  val project                       = ProjectGen.project("org", "proj", base = nxv.base, mappings = ApiMappings("file" -> schemas.files))
-  val deprecatedProject             = ProjectGen.project("org", "proj-deprecated")
-  val projectWithDeprecatedOrg      = ProjectGen.project("org-deprecated", "other-proj")
-  val projectRef                    = project.ref
-  val diskId                        = nxv + "disk"
-  val diskRev                       = ResourceRef.Revision(iri"$diskId?rev=1", diskId, 1)
-  val diskId2                       = nxv + "disk2"
-  val file1                         = nxv + "file1"
-  val file2                         = nxv + "file2"
-  val file1Encoded                  = UrlUtils.encode(file1.toString)
-  val generatedId                   = project.base.iri / uuid.toString
+  val uuid                     = UUID.fromString("8249ba90-7cc6-4de5-93a1-802c04200dcc")
+  val uuid2                    = UUID.fromString("12345678-7cc6-4de5-93a1-802c04200dcc")
+  val ref                      = Ref.of[Task, UUID](uuid).accepted
+  implicit val uuidF: UUIDF    = UUIDF.fromRef(ref)
+  val org                      = Label.unsafe("org")
+  val orgDeprecated            = Label.unsafe("org-deprecated")
+  val project                  = ProjectGen.project("org", "proj", base = nxv.base, mappings = ApiMappings("file" -> schemas.files))
+  val deprecatedProject        = ProjectGen.project("org", "proj-deprecated")
+  val projectWithDeprecatedOrg = ProjectGen.project("org-deprecated", "other-proj")
+  val projectRef               = project.ref
+  val diskId                   = nxv + "disk"
+  val diskRev                  = ResourceRef.Revision(iri"$diskId?rev=1", diskId, 1)
+  val diskId2                  = nxv + "disk2"
+  val file1                    = nxv + "file1"
+  val file2                    = nxv + "file2"
+  val fileTagged               = nxv + "fileTagged"
+  val fileTagged2              = nxv + "fileTagged2"
+  val file1Encoded             = UrlUtils.encode(file1.toString)
+  val generatedId              = project.base.iri / uuid.toString
+  val generatedId2             = project.base.iri / uuid2.toString
 
   val content = "file content"
   val path    = AbsolutePath(JavaFiles.createTempDirectory("files")).rightValue
   val digest  =
     ComputedDigest(DigestAlgorithm.default, "e0ac3601005dfa1864f5392aabaf7d898b1b5bab854f1acb4491bcd806b76b0c")
 
-  def attributes(filename: String = "file.txt", size: Long = 12): FileAttributes = FileAttributes(
-    uuid,
-    s"file://$path/org/proj/8/2/4/9/b/a/9/0/$filename",
-    Uri.Path(s"org/proj/8/2/4/9/b/a/9/0/$filename"),
-    filename,
-    Some(`text/plain(UTF-8)`),
-    size,
-    digest,
-    Client
-  )
+  def withUUIDF[T](id: UUID)(test: => T): T = (for {
+    old <- ref.getAndSet(id)
+    t   <- Task.delay(test).onError(_ => ref.set(old))
+    _   <- ref.set(old)
+  } yield t).accepted
+
```

Comment:
I'm guessing this is because of the exact-matching and lack of fake storage? Maybe there's another way? Perhaps not asserting on the UUID is a possible

Reply:
Yeah and wanted to avoid having to rewrite all the other tests. It is possible to just use the returned one, reckon that would be preferable? Currently it does verify at least that the UUID generator is called, and that it's passed out on the response

Reply:
Let's maybe have a chat IRL because I think I might be misunderstanding what's going on here